docs/source/custom_sockets.rst

```diff
@@ -0,0 +1,244 @@
+.. _custom-sockets:
+
+=========================
+Custom Sockets in Syncano
+=========================
+
+``Syncano`` gives its users an ability to create Custom Sockets. What it means is that users can define 
+a very specific endpoints in their Syncano application, and use them exactly like they would any other Syncano 
+module (Classes, Scripts, etc), using standard API calls.
+Currently, Custom Sockets allow only one dependency - Scripts. It means that under the hood,
+each API call executes a Script, and result of this execution is returned as a result of the
+API call.
+
+Creating a custom socket
+------------------------
+
+To create a custom socket follow these steps::
+
+    import syncano
+    from syncano.models import CustomSocket, Endpoint, ScriptCall, ScriptDependency, RuntimeChoices
+    from syncano.connection import Connection
+
+    # 1. Initialize a custom socket.
+    custom_socket = CustomSocket(name='my_custom_socket')  # this will create an object in place (do API call)
+
+    # 2. Define endpoints.
+    my_endpoint = Endpoint(name='my_endpoint')  # no API call here
```

Comment:
An Endpoint helper - does not validate name in any way. If backend will handle this - it should work (maybe some little changes will be needed). Or I am missing the point?